libsql/src/wasm/mod.rs

```diff
@@ -73,15 +77,84 @@ where
             statements.push(crate::hrana::proto::Stmt::new(s.stmt, false));
         }
         self.conn
-            .raw_batch(statements)
+            .batch_inner(statements)
             .await
             .map_err(|e| crate::Error::Hrana(e.into()))?;
         Ok(())
     }
 
     pub async fn query(&self, sql: &str, params: impl IntoParams) -> crate::Result<Rows> {
         tracing::trace!("querying `{}`", sql);
-        let mut stmt = crate::hrana::Statement::new(self.conn.clone(), sql.to_string(), true);
+        let mut stmt =
+            crate::hrana::Statement::from_connection(self.conn.clone(), sql.to_string(), true);
+        stmt.query(&params.into_params()?).await
+    }
+
+    pub async fn transaction(
+        &self,
+        tx_behavior: TransactionBehavior,
+    ) -> crate::Result<Transaction<T>> {
+        let stream = self.conn.open_stream();
+        let tx = HttpTransaction::open(stream, tx_behavior)
+            .await
+            .map_err(|e| crate::Error::Hrana(e.into()))?;
+        Ok(Transaction { inner: tx })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Transaction<T>
+where
+    T: for<'a> HttpSend<'a>,
+{
+    inner: HttpTransaction<T>,
+}
+
+impl<T> Transaction<T>
+where
+    T: for<'a> HttpSend<'a>,
+{
+    pub async fn query(&self, sql: &str, params: impl IntoParams) -> crate::Result<Rows> {
+        tracing::trace!("querying `{}`", sql);
```

Comment:
nit: could prefix the logs here like so to make it easier to debug in the future.
```suggestion
        tracing::trace!("txn querying `{}`", sql);
```